src/main/java/database/models/FormplayerCaseIndexTable.java

```diff
@@ -131,38 +131,29 @@ public void clearCaseIndices(Collection<Integer> idsToClear) {
         String[] projection = new String[] {COL_CASE_RECORD_ID, COL_INDEX_TARGET, COL_INDEX_RELATIONSHIP};
         HashMap<Integer,Vector<Pair<String, String>>> caseIndexMap = new HashMap<>();
 
-        PreparedStatement selectStatement = null;
-        try {
-            selectStatement = SqlHelper.prepareTableSelectProjectionStatement(connectionHandler.getConnection(),
+        try (PreparedStatement selectStatement = SqlHelper.prepareTableSelectProjectionStatement(connectionHandler.getConnection(),
                     TABLE_NAME,
-                    projection);
-            ResultSet resultSet = selectStatement.executeQuery();
-            while (resultSet.next()) {
-                int caseRecordId = resultSet.getInt(resultSet.findColumn(COL_CASE_RECORD_ID));
-                String targetCase = resultSet.getString(resultSet.findColumn(COL_INDEX_TARGET));
-                String relationship = resultSet.getString(COL_INDEX_RELATIONSHIP);
-                Pair<String, String> index  = new Pair<> (targetCase, relationship);
-
-                Vector<Pair<String, String>> indexList;
-                if (!caseIndexMap.containsKey(caseRecordId)) {
-                    indexList = new Vector<>();
-                } else {
-                    indexList = caseIndexMap.get(caseRecordId);
+                    projection)) {
+            try (ResultSet resultSet = selectStatement.executeQuery()) {
```

Comment:
I could be confused, but instead of nesting `try`, could you use two resources in one try-with-resource?

```
try (
    PreparedStatement selectStatement = SqlHelper.prepareTableSelectProjectionStatement(
        connectionHandler.getConnection(), TABLE_NAME, projection);
    ResultSet resultSet = selectStatement.executeQuery()
) {
    ...
} catch(...) {
    ...
}
```

I'm getting this from the example towards the bottom of https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

There are about a dozen places in this PR where this comment might apply, but I'll let you figure out if it's even a good suggestion

Reply:
You can declare more than one resource in the block., yeah. I didn't do it often since there were a few places where we were looping one of the objects or it wasn't super obvious whether intermediate code could be re-arranged. 

I think it would be the better pattern of the two in a few places, and will try to come back and apply if I have time.